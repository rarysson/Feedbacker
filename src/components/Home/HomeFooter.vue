<template>
  <div class="felx justify-center py-10 bg-brand-gray">
    <p class="font-medium text-center text-gray-800">feedbacker Â© 2020</p>
  </div>
</template>

<script>
export default {
  name: "HomeFooter",
};
</script>
