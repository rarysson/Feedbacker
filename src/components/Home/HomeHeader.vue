<template>
  <header class="header">
    <div class="header-group">
      <div class="flex justify-between items-center py-10">
        <div class="w-28 lg:w-36">
          <img class="w-full" src="@/assets/image/logo.png" alt="logo" />
        </div>
        <div class="flex">
          <button
            class="
              px-6
              py-2
              font-bold
              rounded-full
              text-white
              focus:outline-none
            "
            data-test="create-account-header-btn"
            @click="emit('create-account')"
          >
            Criar uma conta
          </button>
          <button
            class="
              px-6
              py-2
              font-bold
              bg-white
              rounded-full
              text-brand-main
              focus:outline-none
            "
            data-test="login-btn"
            @click="emit('login')"
          >
            Entrar
          </button>
        </div>
      </div>
      <div class="flex flex-col mt-28">
        <h1 class="text-4xl font-black text-white">
          Tenha um feedback.
          <br class="hidden lg:inline-block" />
          E faça seus clientes mais
          <br class="hidden lg:inline-block" />
          felizes!
        </h1>
        <p class="text-lg font-medium text-white">
          Receba ideias, reclamações e feedbacks com um
          <br class="hidden lg:inline-block" />
          simples widget na página.
        </p>
        <div>
          <button
            class="
              px-6
              py-2
              mt-10
              font-bold
              bg-white
              rounded-full
              text-brand-main
              focus:outline-none
            "
            data-test="create-account-central-btn"
            @click="emit('create-account')"
          >
            Criar uma conta grátis
          </button>
        </div>
      </div>
    </div>
  </header>
</template>

<script>
export default {
  name: "HomeHeader",

  setup(_, { emit }) {
    return {
      emit,
    };
  },
};
</script>

<style lang="postcss" scoped>
.header {
  @apply bg-brand-main w-full flex justify-center;
  height: 700px;
}

.header-group {
  @apply flex flex-col w-4/5 max-w-6xl;
}

@media (min-width: 640px) {
  .header-group {
    background-image: url("~@/assets/image/blue_balloons.png");
    background-size: 628px;
    background-position: 90% 100%;
    background-repeat: no-repeat;
  }
}
</style>
