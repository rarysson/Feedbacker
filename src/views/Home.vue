<template>
  <home-header @login="handleLogin" @create-account="handleCreateAccount" />
  <contact />
  <home-footer />
</template>

<script>
import { onMounted } from "@vue/runtime-core";
import { useRoute } from "vue-router";

import Contact from "../components/Home/Contact.vue";
import HomeHeader from "../components/Home/HomeHeader.vue";
import HomeFooter from "../components/Home/HomeFooter.vue";

export default {
  name: "Home",

  components: {
    Contact,
    HomeHeader,
    HomeFooter,
  },

  setup() {
    const router = useRoute();

    onMounted(() => {
      const token = window.localStorage.getItem("token");

      if (token) {
        router.push({ name: "Feedbacks" });
      }
    });

    function handleLogin() {}

    function handleCreateAccount() {}

    return {
      handleLogin,
      handleCreateAccount,
    };
  },
};
</script>
