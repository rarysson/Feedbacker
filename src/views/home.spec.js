import Home from "./Home.vue";
import { routes } from "../router";
import { shallowMount } from "@vue/test-utils";
import { createRouter, createWebHistory } from "vue-router";

const router = createRouter({
  history: createWebHistory("/"),
  routes,
});

describe("Home", () => {
  it("deveria conseguir renderizar Home corretamente", async () => {
    router.push("/");
    await router.isReady();
    const wrapper = shallowMount(Home, {
      global: {
        plugins: [router],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
