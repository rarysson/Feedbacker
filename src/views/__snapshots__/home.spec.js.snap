// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home deveria conseguir renderizar Home corretamente 1`] = `
<home-header-stub></home-header-stub>
<contact-stub></contact-stub>
<home-footer-stub></home-footer-stub>
<modal-factory-stub></modal-factory-stub>
`;
